### PR TITLE
Fix conversion of chained logical operators

### PR DESCRIPTION
**Pull request by Jezurko**
`a && (!b || !c)`

### CODE REVIEW

File: lib/vast/Conversion/Core/ToLLVM.cpp [hunk 1]
```diff
@@ -45,8 +47,10 @@ namespace vast
 
                 auto &first_block = lazy_region.front();
                 // The rewriter API doesn't provide a call to insert into a selected block
+                auto target_it = target->getIterator();
+                std::advance(target_it, 1);
```
Comment (xlauko) [inline lib/vast/Conversion/Core/ToLLVM.cpp:51]: use directly 

```
                auto target_it = std::next(target->getIterator());
```

File: lib/vast/Conversion/Core/ToLLVM.cpp [hunk 2]
```diff
@@ -104,7 +108,12 @@ namespace vast
 
                 cond_br_lhs(rewriter, op.getLoc(), cmp_lhs, rhs_block, end_block);
 
-                rewriter.setInsertionPointToEnd(rhs_block);
+                // In the case that `rhs` consists of multiple blocks we need to
+                // insert to the last block…
```
Comment (xlauko) [inline lib/vast/Conversion/Core/ToLLVM.cpp:112]: ...?

File: lib/vast/Conversion/Core/ToLLVM.cpp [hunk 3]
```diff
@@ -104,7 +108,12 @@ namespace vast
...
+                // e.g.: in a && (b || c) mlir first matches the `||` which creates
+                // it's own cf blocks and then matches the `&&`
+                auto rhs_end_it = std::prev(end_block->getIterator(), 1);
```
Comment (xlauko) [inline lib/vast/Conversion/Core/ToLLVM.cpp:115]: no need of `1` its implicit argument of `std::prev`

File: lib/vast/Conversion/Core/ToLLVM.cpp [hunk 4]
```diff
@@ -104,7 +108,12 @@ namespace vast
...
+                rewriter.setInsertionPointToEnd(&*rhs_end_it);
```
Comment (xlauko) [inline lib/vast/Conversion/Core/ToLLVM.cpp:116]: does this match [getSinglePredecessor](https://mlir.llvm.org/doxygen/classmlir_1_1Block.html#a72289303038da4c089aeb47ca97c29ac)?

File: lib/vast/Conversion/Core/ToLLVM.cpp (hunk 2 above)
Comment (Jezurko) [reply lib/vast/Conversion/Core/ToLLVM.cpp]: I tried to update the comment. Is it now explained better or should I try more?

**Review by xlauko — APPROVED** [review]

**Pull request merged**